WMF Framework/WMFExploreFeedContentController.m

```diff
@@ -70,6 +69,15 @@ - (void)setOperationQueue:(NSOperationQueue *)operationQueue {
 - (void)setDataStore:(MWKDataStore *)dataStore {
     _dataStore = dataStore;
     self.exploreFeedPreferencesUpdateCoordinator = [[ExploreFeedPreferencesUpdateCoordinator alloc] initWithFeedContentController:self];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(viewContextDidSave:) name:NSManagedObjectContextDidSaveNotification object:self.dataStore.viewContext];
+}
+
+- (NSDictionary *)exploreFeedPreferences {
+    if (!_exploreFeedPreferences) {
+        assert([NSThread isMainThread]);
```

Comment:
This assertion either needs to be moved up or this method (and the setter) needs to be made atomic. Otherwise you could be accessing this at the same time on two different threads.

Reply:
If the assertion moves up, the setter would also need a main thread assertion